### ISSUE TITLE
GlobalVariablesSniff infinite loop when missing ; (T_SEMICOLON)

### ISSUE THREAD

**Issue by vaurdan** [WordPress-Coding-Standards/WordPress-Coding-Standards#850]
I was running PHPCS on a project and I noticed that for some reason, a specific file was not being scanned and after a while my computer fans were at 100%. I looked closely and there was an hanging PHP process, which I killed. 

When I ran that PHPCS on the terminal, I got the following output:

```terminal
$ phpcs --standard=WordPress file.php
Notice: Undefined offset: 102 in /Users/henrique/phpcs/wpcs/WordPress/Sniffs/Variables/GlobalVariablesSniff.php on line 325

Notice: Undefined offset: 103 in /Users/henrique/phpcs/wpcs/WordPress/Sniffs/Variables/GlobalVariablesSniff.php on line 325

Notice: Undefined offset: 104 in /Users/henrique/phpcs/wpcs/WordPress/Sniffs/Variables/GlobalVariablesSniff.php on line 325

(...)
```

The `WordPress_Sniffs_Variables_GlobalVariablesSniff` was stuck on an infinite loop and it was because the global variable in the scanned file does not end with a `;`:

```php
<?php global $my_variable ?>
```

I have made this quick fix and I think it works. If you think this is the correct approach, I can make a PR:

```diff
diff --git a/WordPress/Sniffs/Variables/GlobalVariablesSniff.php b/WordPress/Sniffs/Variables/GlobalVariablesSniff.php
index c243b7e..9cecc29 100644
--- a/WordPress/Sniffs/Variables/GlobalVariablesSniff.php
+++ b/WordPress/Sniffs/Variables/GlobalVariablesSniff.php
@@ -321,6 +321,12 @@ class WordPress_Sniffs_Variables_GlobalVariablesSniff extends WordPress_Sniff {
                        $ptr = ( $stackPtr + 1 );
                        while ( $ptr ) {
                                $ptr++;
+
+                               // Halt if there is no semicolon and there is no more tokens to process
+                               if ( ! isset( $this->tokens[ $ptr ] ) ) {
+                                       $ptr = false;
+                               }
+
                                $var = $this->tokens[ $ptr ];
                                if ( T_VARIABLE === $var['code'] ) {
                                        $varname = substr( $var['content'], 1 );
```

**Comment by jrfnl** [comment]
@vaurdan Thanks for reporting this, but I believe this was already fixed in master by PR #736. Could you pelase test your code against the WPCS `develop` branch to confirm ?

**Comment by vaurdan** [comment]
Oops, for some reason I didn't found that when I looked for similar issues. It solves it :)

Closing.
